lms/djangoapps/course_goals/models.py

```diff
@@ -0,0 +1,32 @@
+"""
+Course Goals Models
+"""
+from django.contrib.auth.models import User
+from django.db import models
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from eventtracking import tracker
+from openedx.core.djangoapps.xmodule_django.models import CourseKeyField
+
+
+class CourseGoal(models.Model):  # pylint: disable=model-missing-unicode
+    """
+    Represents a course goal set by a user on the course home page.
+    """
+    user = models.ForeignKey(User)
+    course_key = CourseKeyField(max_length=255, db_index=True, blank=True)
```

Comment:
Why are you indexing on this field? Will you ever query this data solely by course key? Why is this field allowed to be blank?

Reply:
Yes, was thinking it would be beneficial to have the ability to look at the course goal breakdown for a particular course. Removing blank.